ios/Classes/Modules/ReaderModule.swift

```
@@ -0,0 +1,110 @@
+        case .banned:
+            errorMessage = "This device is banned from using the Tap To Pay reader."
+        case .linkingFailed:
+            errorMessage = "The Tap To Pay reader could not link/relink using the provided Apple ID."
+        case .linkingCanceled:
+            errorMessage = "User has canceled the linking/relinking operation."
+        case .invalidToken:
+            errorMessage = "The Tap To Pay reader generated an invalid token."
+        case .notAuthorized:
+            errorMessage = "This device must be authorized with a Square account in order to use Tap To Pay."
+        case .notAvailable:
+            errorMessage = "The Tap To Pay reader is not available on this device or device's operating system."
+        case .noNetwork:
+            errorMessage = "The Tap To Pay reader could not connect to the network. Please reconnect to the Internet and try again."
+        case .networkError:
+            errorMessage = "The network responded with an error."
+        case .other:
+            errorMessage = "An error with the Tap To Pay reader has occurred. Please try again."
+        case .passcodeDisabled:
+            errorMessage = "This device does not currently have an active passcode set."
+        case .unexpected:
+            errorMessage = "Mobile Payments SDK encountered an unexpected error. Please try again."
+        case .unsupportedOSVersion:
+            errorMessage = "The device's OS version does not meet the minimum requirement of iOS 16.7 for Tap to Pay on iPhone."
+        case .unsupportedDeviceModel:
+            errorMessage = "This device model is not currently supported to use the Tap To Pay reader."
+        default:
+            errorMessage = defaultError
+        }
+        return errorMessage
+    }
+
+
+    public static func showMockReaderUI(result: @escaping FlutterResult) {
+        do {
+            try mockReader?.present()
+        } catch  {
+
```

we should propagate the error the SDK returns in here

---

i propagate the error to sdk